analyses/independent-samples-release/util/independent-rna-samples.R

```diff
@@ -0,0 +1,118 @@
+  
+  
+  
+  # Find sample set for the dna independent samples 
+  # This will always be the included since in both the following
+  # conditions "independent_dna" "independent_dna_plus_only_rna"
+  # 
+  independent_dna <- histology_df %>%
+    dplyr::mutate(match_id = paste(Kids_First_Participant_ID, sample_id, sep = "_")) %>%
+    # include matched independent_dna samples
+    dplyr::filter(Kids_First_Biospecimen_ID %in%
+                    independent_dna_sample_df$Kids_First_Biospecimen_ID)
+  matched_rna <- histology_df %>%
+    dplyr::mutate(match_id = paste(Kids_First_Participant_ID, sample_id, sep = "_"),
+                  age_at_diagnosis_days = as.numeric(age_at_diagnosis_days)) %>%
+    # keep rna from histology_df
+    dplyr::filter(experimental_strategy == "RNA-Seq",
+                  # find participants which have matching dna samples in independent_wgswxs
+                  Kids_First_Participant_ID %in% independent_dna$Kids_First_Participant_ID,
+                  # keep specific sample_ids since some participants might have multiple sample_ids
+                  match_id %in% independent_dna$match_id)
+  
+  # has rna samples which match the independent samples provided
+  sample_df <- matched_rna
+
+  
+  # Here we are adding only initial only-RNA-Seq samples
+  # since this will always to part of independent_dna_plus_only_rna
+  # regardless tumor_description_rna_only is "primary" OR "primary_plus"
+  #
+  if( match_type == "independent_dna_plus_only_rna" ) {
+    # find sample set where we initial only-RNA-Seq samples
+    only_rna_intial <- histology_df %>% 
+      # keep rna from histology_df
+      dplyr::filter(experimental_strategy == "RNA-Seq",
+                    tumor_descriptor %in% primary_descs,
+                    # find and remove participants which have 
+                    # matching dna samples in independent_wgswxs 
+                    !Kids_First_Participant_ID %in% independent_dna$Kids_First_Participant_ID)
+    
+    # has rna samples which match the independent samples provided plus rna only sample which are primary tumors
+    sample_df <- bind_rows(sample_df,only_rna_intial)
+  }
+  
+  
+  # Here we are adding only-RNA-Seq samples which are not initial
+  # if tumor_description_rna_only == "primary_plus"
+  #
+  if(match_type == "independent_dna_plus_only_rna" & tumor_description_rna_only == "primary_plus"){
+    # find sample set where we only find rna samples
+    only_rna_plus <- histology_df %>% 
+      # keep rna from histology_df
+      dplyr::filter(experimental_strategy == "RNA-Seq",
+                    # find and remove participants which have 
+                    # matching dna samples in independend_wgswxs 
+                    !Kids_First_Participant_ID %in% independent_dna$Kids_First_Participant_ID,
+                    # and participant not in only_rna_initial sample set
+                    !Kids_First_Participant_ID %in% only_rna_intial$Kids_First_Participant_ID
```

Comment:
```suggestion
                    # find and remove participants which have 
                    # matching dna samples in independend_wgswxs 
                    # or primary RNA-seq samples.
                    !Kids_First_Participant_ID %in% sample_df$Kids_First_Participant_ID
```

I am not sure about this suggestion either.